File: backend/src/main/java/com/farid/backend/dto/UserDTO.java
package com.farid.backend.dto;

import lombok.Builder;
import lombok.Data;

import java.util.UUID;
@Data
@Builder
public class UserDTO {
    private UUID id;
    private String name;
    private String email;
    private String password;
    private String phone;
    private String userImage;
    private AddressDTO addressDTO;
    private CartDTO cartDTO;
}

File: frontend/src/app/model/UserDTO.ts
import { AddressDTO } from "./AddressDTO";

export class UserDTO{
    id: string | null;
    name: string;
    email: string;
    phone: string;
    password: string;
    userImage: string;
    addressDTO: AddressDTO;
    cartDTO: any;
    constructor(id: string, name: string, email: string, phone: string, password: string, userImage: string, addressDTO: AddressDTO, cartDTO: any){
        this.id = id;
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.password = password;
        this.userImage = userImage;
        this.addressDTO = addressDTO;
        this.cartDTO = cartDTO;
    }
}